pkg/serving/config_changes.go

```diff
@@ -67,3 +67,11 @@ func UpdateImage(config *servingv1alpha1.ConfigurationSpec, image string) error
 	config.RevisionTemplate.Spec.Container.Image = image
 	return nil
 }
+
+func UpdateResources(config *servingv1alpha1.ConfigurationSpec, requestsResourceList corev1.ResourceList, limitsResourceList corev1.ResourceList) error {
+	config.RevisionTemplate.Spec.Container.Resources = corev1.ResourceRequirements{
```

Comment:
here is potentially better way to do it:

instead of reassigning entire `config.RevisionTemplate.Spec.Container.Resources`, update them selectively:

```
for k, v := range requestsResourceList  {
  config.RevisionTemplate.Spec.Container.Resources.Requests[k] = v
}
```

(+ also initialize ResourceList map since it could be empty)

this would simplify `compute*Resources` methods as they would only have to return additions.

Reply:
Next commit addresses this